### PR TITLE
blog: April community gems

### PR DESCRIPTION
**Pull request by flippedcoder**
Tentative publish date: 04/28/22

Things left to update:
- [x] image (make sure to use https://tinypng.com/)
- [x] comments link
- [x] May office hours description/link
- [x] technical style (code blocks formatted correctly with dvc, python, yaml, etc)

### CODE REVIEW

File: content/blog/2022-04-26-april-22-community-gems.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: April '22 Community Gems
+date: 2022-04-26
+description: >
+  A roundup of technical Q&A's from the DVC and CML community. This month: CML
+  updates, working with multiple datasets, using DVC stages, and more.
+descriptionLong: >
+  A roundup of technical Q&A's from the DVC and CML community. This month: CML
+  updates, working with multiple datasets, using DVC stages, and more.
```
Comment (casperdcl) [inline content/blog/2022-04-26-april-22-community-gems.md:9]: long/short seems same? (feel free to resolve in intentional).

Also prefer *sans* `'`

```suggestion
  A roundup of technical Q&As from the DVC and CML community. This month: CML
  updates, working with multiple datasets, using DVC stages, and more.
descriptionLong: >
  A roundup of technical Q&As from the DVC and CML community. This month: CML
  updates, working with multiple datasets, using DVC stages, and more.
```

File: content/blog/2022-04-26-april-22-community-gems.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Thanks for the question @Atsu!
+
+This is supported out-of-the-box! Here's how it works:
+
+1. GitHub action launches "self-hosted" GCP or AWS using
+   `cml runner --reuse --labels=cml`
+2. GitHub action runs the rest of the workflow on the "self-hosted" runner using
+   `runs-on: [self-hosted, cml]` and `timeout-minutes: 4320`
+3. If GitHub action is about to timeout, CML will restart the workflow.
```
Comment (casperdcl) [inline content/blog/2022-04-26-april-22-community-gems.md:67]:
```suggestion
1. Within Github Actions, CML launches a [self-hosted runner](https://cml.dev/doc/self-hosted-runners) on GCP or AWS using
   `cml runner --labels=cml --cloud=gcp`/`--cloud=aws`
2. GitHub Actions runs the rest of the workflow on the self-hosted runner using
   `runs-on: [self-hosted, cml]` and the maximum allowable `timeout-minutes: 4320`
3. If GitHub Actions is about to timeout, CML will restart the workflow,
   so make sure your code regularly caches and restores data if it's expected to
   take >3 days to run.
```

File: content/blog/2022-04-26-april-22-community-gems.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+You can follow along with
+[this doc](https://cml.dev/doc/self-hosted-runners?tab=GitHub#allocating-cloud-compute-resources-with-cml)
+to get started.
+
+The key is requesting GitHub's maximum `timeout-minutes: 4320`. This signals to
```
Comment (casperdcl) [inline content/blog/2022-04-26-april-22-community-gems.md:73]:
```suggestion
The key is requesting GitHub's [maximum `timeout-minutes: 4320`](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#usage-limits).
This signals to
```

File: content/blog/2022-04-26-april-22-community-gems.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+CML to restart the workflow just before timeout. You'll also have to write your
```
Comment (casperdcl) [inline content/blog/2022-04-26-april-22-community-gems.md:74]:
```suggestion
CML to [restart the workflow](https://cml.dev/doc/ref/runner#faqs-and-known-issues) just before the timeout. You'll also have to write your
```

**Review by casperdcl — COMMENTED** [review]
I so referenced

File: content/blog/2022-04-26-april-22-community-gems.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+### [Is `dvc dag` based on `deps` and `outs`, so that a stage that depends on the output of another stage will always be executed after the former has finished?](https://discord.com/channels/485586884165107732/563406153334128681/956113493155799070)
+
+This is a good question from @johnysku!
+
+That is correct! If the pipelines are independent or the stages are independent,
+they may run in any order. Without explicit dependency linkage, stages could be
+executed in an unexpected order.
+
+One way around this is by outputing some kind of placeholder file when all a
+stage is finished and make that the dependency that triggers the next stage.
```
Comment (dberenbaum) [inline content/blog/2022-04-26-april-22-community-gems.md:37]: I'm not fully understanding this description. Maybe it's unnecessary. It sounds like the discord discussion went deeper than the initial question into some specific scenario, but it's not really clear how this relates to the original question.

File: content/blog/2022-04-26-april-22-community-gems.md (hunk 2 above)
Comment (flippedcoder) [reply content/blog/2022-04-26-april-22-community-gems.md:67]: Thanks for this! That actually helped me understand how CML works even better.

File: content/blog/2022-04-26-april-22-community-gems.md (hunk 5 above)
Comment (flippedcoder) [reply content/blog/2022-04-26-april-22-community-gems.md]: Gotcha, thanks for catching this.

**Pull request merged**